src/ast/ast_validations.c

```diff
@@ -179,34 +179,169 @@ static AST_Validation _Validate_MATCH_Clause_ReusedEdges(const cypher_astnode_t
     return AST_VALID;
 }
 
-static AST_Validation _Validate_MATCH_Clause_Relations(const cypher_astnode_t *node, TrieMap *identifiers, char **reason) {
-    cypher_astnode_type_t node_type = cypher_astnode_type(node);
+static AST_Validation _ValidatePath(const cypher_astnode_t *path, TrieMap *identifiers, char **reason) {
+	AST_Validation res = AST_VALID;
+	uint path_len = cypher_ast_pattern_path_nelements(path);
+
+	// Check all entities on the path
+	for (uint i = 0; i < path_len; i ++) {
```

Comment:
if we only care about odd indices 
`for (uint i = 1; i < path_len; i +=2) {`
now the `if(1 % 2)` can be omitted.